### PR TITLE
Allow filtering public/private IPv6 addresses in ip_addrs6

### PR DESCRIPTION
**Pull request by rmcintosh**
* Allow filtering public/private IPv6 addresses in ip_addrs6
* Clean up the ip_addrs funcs to match the ip_addrs6 funcs

I'm totally fine with backing out the clean up I did in ip_addrs, though IMO it's a bit clearer/cleaner as I have it here.

### What does this PR do?
This PR allows filtering private/public addresses when calling ip_addrs6 from network/win_network, in the same way that the ip_addrs funcs allow doing this.

### What issues does this PR fix or reference?
Fixes: always getting e.g. link-local v6 addresses returned in ip_addrs6 output.

### Previous Behavior
Previously the output would always include e.g. link-local v6 addresses, which (from my POV) is probably not useful for most users.

### New Behavior
The output is the same by default, but if you pass the new "type" argument, you can get output with just the specified type of addresses.

### Merge requirements satisfied?
- [x] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: salt/modules/network.py [hunk 1]
```diff
@@ -1322,10 +1323,17 @@ def ip_addrs6(interface=None, include_loopback=False, cidr=None):
     addrs = __utils__["network.ip_addrs6"](
         interface=interface, include_loopback=include_loopback
     )
+
     if cidr:
         return [i for i in addrs if __utils__["network.in_subnet"](cidr, [i])]
-    else:
-        return addrs
+
+    if type == "public":
```
Comment (rmcintosh) [inline salt/modules/network.py:1330]: alternately, we can define the lists of addresses inside of each if statement and then have a single return at the bottom (for a function of this length/complexity, i'm not sure it matters, though)

**Comment by krionbsd** [comment]
@rmcintosh thank you for contribution! Could you please run `pre-commit -av` and re-commit the changes? Could you please add a test case for this change?

**Comment by rmcintosh** [comment]
@krionbsd I think I managed to wrangle the pre-commit - let me know if I somehow botched it (this is my first salt contribution when using that tool).

I'll get to work on adding a testcase 👍

**Comment by krionbsd** [comment]
> @krionbsd I think I managed to wrangle the pre-commit - let me know if I somehow botched it (this is my first salt contribution when using that tool).
> 
> I'll get to work on adding a testcase 👍

That's awesome @rmcintosh ! Thank you so much!

**Comment by rmcintosh** [comment]
@krionbsd I added a test case which accounts for calling the function with `type` set to each of `private`/`public`, as well as without the argument supplied 👍 let me know how it looks!

File: tests/unit/modules/test_network.py [hunk 2]
```diff
@@ -228,6 +225,36 @@ def test_ip_addrs6(self):
         ):
             self.assertListEqual(network.ip_addrs6("int", "include"), ["A"])
 
+    def test_ip_addrs6_type(self):
+        """
+        Test for returns a list of IPv6 addresses assigned to the host filtered by type.
+        """
+        ipv6_addrs = ["fe80::f03c:91ff:fef1:21e3", "2600:3c03::f03c:91ff:fef1:21e3"]
```
Comment (rmcintosh) [inline tests/unit/modules/test_network.py:232]: I grabbed these examples from a random old machine of mine as python's IP address validation did not like purely example addresses that I had tried

**Review by krionbsd — APPROVED** [review]

**Review by krionbsd — DISMISSED** [review]

**Comment by rmcintosh** [comment]
I noticed some conflicts have arisen on this PR in the past couple of weeks - I assume you all would like me to get those fixed up for this to move forward?

**Comment by rmcintosh** [comment]
conflicts resolved 👍

**Review by krionbsd — DISMISSED** [review]

**Comment by rmcintosh** [comment]
fixed the merge conflict on this one 👍🏽

**Comment by rmcintosh** [comment]
Still willing to do anything necessary to get this one over the finish line 🙂

**Comment by Ch3LL** [comment]
Looks like pre-commit is failing. Once its passing I'll give this a review.

**Comment by rmcintosh** [comment]
@Ch3LL thank you for pointing that out and for your time 🙇🏽 I think I have resolved the pre-commit errors, please let me know if you notice anything else 👍🏽

File: salt/modules/network.py [hunk 3]
```diff
@@ -1291,25 +1291,27 @@ def ip_addrs(interface=None, include_loopback=False, cidr=None, type=None):
...
-def ip_addrs6(interface=None, include_loopback=False, cidr=None):
+def ip_addrs6(interface=None, include_loopback=False, cidr=None, type=None):
     """
     Returns a list of IPv6 addresses assigned to the host. ::1 is ignored,
     unless 'include_loopback=True' is indicated. If 'interface' is provided,
     then only IP addresses from that interface will be returned.
     Providing a CIDR via 'cidr="2000::/3"' will return only the addresses
-    which are within that subnet.
+    which are within that subnet. If 'type' is 'public', then only public
+    addresses will be returned. Ditto for 'type'='private'.
```
Comment (Ch3LL) [inline salt/modules/network.py:1314]: Can you add a versionadded here for this new arg for the version 3007.0.

File: salt/modules/win_network.py [hunk 4]
```diff
@@ -432,6 +433,10 @@ def ip_addrs6(interface=None, include_loopback=False, cidr=None):
 
         .. versionchanged:: 2019.2.0
 
+    type
+        If option set to 'public' then only public addresses will be returned.
```
Comment (Ch3LL) [inline salt/modules/win_network.py:437]: same here, can you add a versionadded for 3007.0 for this new option.

**Review by Ch3LL — CHANGES_REQUESTED** [review]
Also it looks like you deleted salt/client/ssh/__init__.py which I believe is a mistake. Can you re-add that?

**Comment by dwoz** [comment]
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

**Pull request closed without merging**